### DISCUSSION TITLE
Force_parameters file and Fmask cloud thresholds

### DISCUSSION THREAD

**Discussion by HertugHelms** [davidfrantz/force#292] [Q&A]
Hi - i am trying to fine-tune the Fmask cloud masking part of Force l2ps and i am using the force_parameters_s2.prm file to do so.

I have been testing different CLOUD_THRESHOLDS both with low values of 0.01 to high values up to 0.5 but it seems like it makes no difference. Force will output the exact same QAI and OVV file for all the tested thresholds.

Anyone have similar experience or have tested themselves and found it to work?

Thanks!

**Comment by davidfrantz** [comment]
Dear @HertugHelms,

I guess you are processing Sentinel-2 images?

The cloud probability threshold is used in translating the cloud probability into a discrete cloud mask (see Zhu & Woodcock, 2012). However, in FORCE, this only applies to Landsat. This is because FORCE completely substitutes the original cloud probability computation with the parallax-based approach described in Frantz et al. 2018.

However, the shadow threshold should affect both data streams. As far as I remember, a lower threshold results in potentially more shadows (though other constraints apply, too). A higher threshold should decrease the amount of shadow.

Best,
David

**Reply by HertugHelms** [reply]
Hi David - Thanks for answering so fast! Yes this is Sentinel-2 image processing on a scene level so 1 .SAFE file at a time. Okay, that helps my sanity a bit that the cloud_threshold isn't being used. Do you have any clues as to how i can more aggressively mask out clouds on S2 then?

There seems to be a certain type of cloud that creates a lot of trouble.

//Casper

**Comment by davidfrantz** [comment]
Hi Casper,

using bigger buffers would be an option.
You could also go into code and change the threshold of the CDI layer.

What kind of clouds are problematic?

**Reply by HertugHelms** [reply]
Front of stratus clouds have shown some problems - specifically the tile 32VNJ on the 20230228 Sentinel 2 image. There the cloud mask doesn't capture a large part of the cloud atleast.
